src/data.py

```diff
@@ -8,258 +8,372 @@
+            filtered_data.append({'doc_id':document['doc_id'], 'text': document['text'], 'annotations':filtered_annotations})
+        self.corpus = Corpus(filtered_data)
+        
+    # def add_ids_to_annotations(self):    
+    #     id =0    
+    #     for document in self.data:
+    #         # id = defaultdict(lambda: 0)
+    #         for ann in document['annotations']:
+    #             id +=1
+    #             ann['ann_id'] = ann['label'][:4] +"_"+ str(id[ann['label']])
     
+    def merge_annoatation(self):
+        total_collisions=defaultdict(lambda: 0)
+        entities  = self.corpus.get_entities()
         
-    return  DocumentReaderDatasetForTraining(dataset=train_data, 
-                                             tokenizer=tokenizer,
-                                             context_size = context_size,
-                                             transforms=train_transformations,
-                                             augmentations=train_augmentations), \
-            DocumentReaderDatasetForTraining(dataset=test_data, 
-                                             tokenizer=tokenizer,
-                                             context_size=context_size,
-                                             transforms=test_transformations)
-
+        for doc in self.corpus:
+            new_annotations = []
+
+            sample_annotations = defaultdict(lambda: RangeDict())
+            new_annotation_index = 0
+            for annotation in doc["annotations"]:
+                # checks for a collision and returns the spans of the new entity
+                new_span = sample_annotations[annotation['label']].maybe_merge_annotations(annotation)
+                if new_span:
+                    new_annotation_index += 1
+                    # lets create a new annotation bc collision
+                    annotation = {"label": annotation['label'], "start_span": new_span[0], "end_span": new_span[1]}
+                    total_collisions[annotation['label']]+=new_span[2]
+                    
+                sample_annotations[annotation['label']][(annotation["start_span"], annotation["end_span"])] = annotation
+            for entity_type in entities:
+                new_annotations.extend(sample_annotations[entity_type].get_all_annotations())
+            doc['annotations'] = new_annotations
+            
+            
+        if sum(total_collisions.values())>0:
+            print(f"Warning, we found {total_collisions} collisions that were automaticly handle by merging strategy")
+        
+    def split_data(self, test_split_percentage):
+        trainCorpus, testCorpus = self.corpus.split(test_split_percentage)
+        return  CorpusPreProcessor(trainCorpus), CorpusPreProcessor(testCorpus)
+    
 
 
-class DocumentReaderDatasetForTraining(torch.utils.data.Dataset):
+def Spanish_Biomedical_NER_Corpus(file_path, documents_folder):
```

Comment:
It's okay, but it would not make more sense to be a class that extends from Corpus? What was the issue here?

Reply:
Indentation might be wrong

```python
class Spanish_Biomedical_NER_Corpus(Corpus):
	def __init__(file_tsv_file, docs_folder):
		annotations = defaultdict(list)
		df = pd.read_csv(file_path, sep="\t")
		for _, row in df.iterrows():
		annotations[row["filename"]].append({k:row[k] for k in ["label", "start_span", "end_span"]})

		#load the documents
		document_text = {}
		for file in annotations.keys():
			with open(os.path.join(documents_folder,f"{file}.txt"), 'r') as f:
				document_text[file] = ''.join([line for line in f]).strip()

		data = [{"doc_id":k, "text":document_text[k], "annotations":v} for k,v in annotations.items()]   

		super().__init__(data)

```